x-pack/plugins/security_solution/server/lib/detection_engine/routes/__mocks__/request_responses.ts

```diff
@@ -738,10 +743,357 @@ export const legacyGetNotificationResult = (): LegacyRuleNotificationAlertType =
+ */
+export const legacyGetDailyNotificationResult = (
+  id = '456',
+  ruleId = '123'
+): LegacyRuleNotificationAlertType => ({
+  id,
+  name: 'Notification for Rule Test',
+  tags: [`__internal_rule_alert_id:${ruleId}`],
+  alertTypeId: 'siem.notifications',
+  consumer: 'siem',
+  params: {
+    ruleAlertId: `${ruleId}`,
+  },
+  schedule: {
+    interval: '1d',
+  },
+  enabled: true,
+  actions: [
+    {
+      group: 'default',
+      params: {
+        message: 'Rule {{context.rule.name}} generated {{state.signals_count}} alerts',
+        to: ['test@test.com'],
+        subject: 'Test Actions',
+      },
+      actionTypeId: '.email',
+      id: '99403909-ca9b-49ba-9d7a-7e5320e68d05',
+    },
+  ],
+  throttle: null,
+  notifyWhen: 'onActiveAlert',
+  apiKey: null,
+  apiKeyOwner: 'elastic',
+  createdBy: 'elastic',
+  updatedBy: 'elastic',
+  createdAt: new Date('2020-03-21T11:15:13.530Z'),
+  muteAll: false,
+  mutedInstanceIds: [],
+  scheduledTaskId: '62b3a130-6b70-11ea-9ce9-6b9818c4cbd7',
+  updatedAt: new Date('2020-03-21T12:37:08.730Z'),
+  executionStatus: {
+    status: 'unknown',
+    lastExecutionDate: new Date('2020-08-20T19:23:38Z'),
+  },
+});
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ */
+export const legacyGetWeeklyNotificationResult = (
+  id = '456',
+  ruleId = '123'
+): LegacyRuleNotificationAlertType => ({
+  id,
+  name: 'Notification for Rule Test',
+  tags: [`__internal_rule_alert_id:${ruleId}`],
+  alertTypeId: 'siem.notifications',
+  consumer: 'siem',
+  params: {
+    ruleAlertId: `${ruleId}`,
+  },
+  schedule: {
+    interval: '7d',
+  },
+  enabled: true,
+  actions: [
+    {
+      group: 'default',
+      params: {
+        message: 'Rule {{context.rule.name}} generated {{state.signals_count}} alerts',
+        to: ['test@test.com'],
+        subject: 'Test Actions',
+      },
+      actionTypeId: '.email',
+      id: '99403909-ca9b-49ba-9d7a-7e5320e68d05',
+    },
+  ],
+  throttle: null,
+  notifyWhen: 'onActiveAlert',
+  apiKey: null,
+  apiKeyOwner: 'elastic',
+  createdBy: 'elastic',
+  updatedBy: 'elastic',
+  createdAt: new Date('2020-03-21T11:15:13.530Z'),
+  muteAll: false,
+  mutedInstanceIds: [],
+  scheduledTaskId: '62b3a130-6b70-11ea-9ce9-6b9818c4cbd7',
+  updatedAt: new Date('2020-03-21T12:37:08.730Z'),
+  executionStatus: {
+    status: 'unknown',
+    lastExecutionDate: new Date('2020-08-20T19:23:38Z'),
+  },
+});
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ */
+export const legacyGetFindNotificationsResultWithSingleHit = (
+  ruleId = '123'
+): FindHit<LegacyRuleNotificationAlertType> => ({
+  page: 1,
+  perPage: 1,
+  total: 1,
+  data: [legacyGetNotificationResult(ruleId)],
```

Comment:
This is passing `ruleId` into `legacyGetNotificationResult` in the parameter for `id` instead of the second parameter `ruleId`, which is why the [tests above](https://github.com/elastic/kibana/pull/130511/files#diff-27acb6aa8e391cb95bb29beeca4a3bf63c2bca678dcd7e2807fccf8a685ee4edR130) are needing to set the `id` in the call to `legacyGetNotificationResult` as well. I'd suggest making `legacyGetFindNotificationsResultWithSingleHit` take (id ='456', ruleId='123') as params as well or use named parameters.

Reply:
Thanks! Updated.